vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/editor/EditorImpl.java

```diff
@@ -216,11 +216,6 @@ private void validate(T item) {
             throw new IllegalStateException("Editing item " + item
                     + " failed. Item editor is already editing item " + edited);
         }
-
-        if (!getGrid().getDataCommunicator().getKeyMapper().has(item)) {
-            throw new IllegalStateException("The item " + item
-                    + " is not in the backing data provider");
-        }
```

Comment:
I guess, it would ideally make a request to the data provider to check if the item exists, but not sure how easy it is to do.